docs/chapters/traversal/index.md

```diff
@@ -0,0 +1,178 @@
+print(len(root.matches))
+>>> 1
+print(root.matches[0].granularity)
+>>> 0
+print(root.matches[0].adjacency)
+>>> 1
+
+```
+
+![adjacency](img/adjacency.png)
+
+In the code snippet and diagram above, we initialized a `Document` as `root` with the text: *What is love? Oh, baby do not hurt me.*.
+And a `Document` with text *What is love? Oh please do not hurt me* was added as a match to the `root`.
+The matched document `match` is a document without any parents, so it stays at the same level as `root` with a granularity value of 0.
+Meanwhile, since `match` is the retrieved result from `root`, so the `adjacency` increased to 1.
+
+By default, the `root` node has an `adjacency` of 0, and the value increases by 1 when it hit a `match`.
+
+## Let's go deeper: Recursive Document Representation
+
+Till now, we've introduced `chunks` and `matches` **with a depth of 1**.
+While in a real-world scenario, things could be much more complicated than this.
+For instance, a `Chunk` could be further divided into small chunks, and a chunk at any level might have it's own `matches` at that level.
+
+![go-deeper](https://hanxiao.io/2020/08/28/What-s-New-in-Jina-v0-5/blog-post-v050-protobuf-documents.jpg)
+
+Jina has defined a recursive structure with **arbitrary width and depth** instead of a trivial bi-level structure.
+Roughly speaking, `chunks` can have the next level chunks and the same level matches; and so do matches.
+This could go on and on. The following figure illustrates this structure [Ref: New Features in Jina v0.5 You Should Know About](https://hanxiao.io/2020/08/28/What-s-New-in-Jina-v0-5/).
+
+![recursive](img/recursive.png)
+
+This recursive structure allows the Document to represent many real-world objects much comfortable than before.
+For example, in NLP a long document is composed of semantic chapters; each chapter consists of multiple paragraphs, which can be further segmented into sentences.
+In CV, a video is composed of one or more scenes, including one or more shots (i.e. a sequence of frames taken by a single camera over a continuous period of time).
+Each shot includes one or more frames. Such hierarchical structures can be very well represented with the recursive representation.
+
+If we look from a tree view (with a depth of 3):
+
+![tree-view](img/tree.png)
+
+## Document Traversal with traversal paths
+
+As you already learned from [Jina 101](https://docs.jina.ai/chapters/101/.sphinx.html), one needs to apply transformation (i.e. a `callback`) on a different level of documents.
+Given the tree structure, how could we achieve that?
+The answer is `traversal`.
+
+Jina has defined a method called `traversal` within the class of `Document`, which looks like this:
+
+```python
+def traverse(self, traversal_path: str, callback_fn: Callable, *args, **kwargs) -> None
+    """Traversal apply `:meth:callback_fn` on the recursive tree representation."""
+    ...
+```
+
+This allow you to apply the `callback_fn` based on `traversal_path`.
```

Comment:
"This allows you.."

Reply:
Either delete "the" in front of `callback_fn` or add "function" after it.

Reply:
updated